runtimes/runtimes/standalone.ts

```diff
@@ -232,6 +239,13 @@ export const standalone = (props: RuntimeProps) => {
                 sendProgress: <P>(type: ProgressType<P>, token: ProgressToken, value: P) => {
                     return lspConnection.sendProgress(type, token, value)
                 },
+                sendEncryptedProgress: async <P>(type: ProgressType<P>, token: ProgressToken, value: P) => {
```

Comment:
Exposing `sendEncryptedProgress` to server implementations means that servers that want to use encryption must use this method instead of plain `sendProgress` in their business logic. 
I think it's not correct, and the condition implemented in this method should be moved to `sendProgress` method itself. This way, encrypted mode will be triggered under the hood, depending if encryption mode was enabled at server start time by client.

Reply:
Not all progress should be encrypted though, only progress for specific types of requests need to be encrypted. 

Right now either approach is not ideal IMO, in the approach that I am proposing we expect the server to be aware that it is sending sensitive progress and explicitly use the encrypted progress. This ensures that only responses that need encryption get encrypted but this also means server implementor has to know about the detail of it


In the alternative approach you are proposing, everything will be encrypted and client would need to decrypt everything. The advantage is that server doesn't need to know or do anything extra

Reply:
> This ensures that only responses that need encryption get encrypted but this also means server implementor has to know about the detail of it

I'm afraid that this detail pushes too much responsibility to individual Capability server implementation. How the server implementation will decide when to send encrypted vs. regular token? Capability server will need to know in which mode server was started if encryption was used or not.
What will happen if Capability calls `sendEncryptedProgress`, but server was started without encryption? The message will be dropped as I read it now.

Why do we need to support encrypted version of `sendProgress` at all? Why can't client send in unencrypted?